tests/lib/state_test.py

```diff
@@ -1293,3 +1295,299 @@ def _create_desired_state():
                 ]
             }
         )
+
+
+class TestLinuxBridgeState:
+    Bridge_name = "bridge0"
+    Port1 = "eth1"
+    Port2 = "eth2"
+
+    def test_add_port_to_bridge(self):
+        current_bridge_state = {
+            Interface.NAME: TestLinuxBridgeState.Bridge_name,
+            Interface.TYPE: InterfaceType.LINUX_BRIDGE,
+            Interface.STATE: InterfaceState.UP,
+            LinuxBridge.CONFIG_SUBTREE: {LinuxBridge.PORT_SUBTREE: []},
+        }
+        current_state = state.State({Interface.KEY: [current_bridge_state]})
+
+        desired_bridge_state = {
+            Interface.NAME: TestLinuxBridgeState.Bridge_name,
+            Interface.TYPE: InterfaceType.LINUX_BRIDGE,
+            Interface.STATE: InterfaceState.UP,
+            LinuxBridge.CONFIG_SUBTREE: {
+                LinuxBridge.PORT_SUBTREE: [
+                    {LinuxBridge.Port.NAME: TestLinuxBridgeState.Port1},
+                ]
+            },
+        }
+        desired_state = state.State({Interface.KEY: [desired_bridge_state]})
+        desired_state.merge_interfaces(current_state)
+
+        desired_bridge_ports = desired_bridge_state.get(
```

Comment:
I think you need to query the `desired_state` object and not on the original row data/dict. They are not the same after the merge (the State constructor copies it).

Reply:
Done.

This exposed a bug on the port merge algorithm - I had to initialize `merged_ports` to the ports coming from the desired state in https://github.com/nmstate/nmstate/pull/888/files#diff-09946053413e41502a142f4bfe77875bR987 

Good catch :)